machines/store.ts

```diff
@@ -114,10 +135,17 @@ export const storeMachine =
             src: 'generateEncryptionKey',
           },
           on: {
-            KEY_RECEIVED: {
-              actions: 'setEncryptionKey',
-              target: 'resettingStorage',
-            },
+            KEY_RECEIVED: [
+              {
+                cond: 'isCustomSecureKeystore',
+                target: 'ready',
+              },
+              {
+                actions: 'setEncryptionKey',
+                target: 'resettingStorage',
```

Comment:
I might be wrong on this one but is the storage getting cleared each time the key is received on a phone which has a hardware keystore? Can this be tested once?